### PR TITLE
Fix missing days exception

### LINKED ISSUE #37: Generating a report for SLI with no days crashes the entire report

### PR DESCRIPTION
**Pull request by mohabusama**
Resolve #37

### CODE REVIEW

File: generate-slr.py [hunk 1]
```diff
@@ -69,8 +70,16 @@ def generate_weekly_report(base_url, product, output_dir):
     # TODO: should use pg_slug from PostgreSQL database (but we don't return it right now)
     product_group = report_data['product']['product_group_name'].lower()
 
-    period_from = min(report_data['service_level_objectives'][0]['days'].keys())[:10]
-    period_to = max(report_data['service_level_objectives'][0]['days'].keys())[:10]
+    period_from = period_to = None
+    for slo in report_data['service_level_objectives']:
+        if len(slo['days']):
```
Comment (hjacobs) [inline generate-slr.py]: To be more pythonic: I would remove the `len(..)` as an empty dict is `False` in Python

File: generate-slr.py [hunk 2]
```diff
@@ -69,8 +70,16 @@ def generate_weekly_report(base_url, product, output_dir):
...
+            period_from = min(slo['days'].keys())[:10]
+            period_to = max(slo['days'].keys())[:10]
+            break
+
+    if not all([period_from, period_to]):
```
Comment (hjacobs) [inline generate-slr.py]: `not period_from or not period_to` sounds easier to read for me

Comment (mohabusama) [reply generate-slr.py]: Done

File: generate-slr.py (hunk 1 above)
Comment (mohabusama) [reply generate-slr.py]: Done

**Comment by hjacobs** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Pull request merged**